docs/l3/api_flow.md

```diff
@@ -0,0 +1,413 @@
+
+Response:
+```
+[
+  {
+    "id": "8c783421-eb61-4bdb-b719-aee945d02121",
+    "state": "finalised",
+    "localId": "18f832d2-162f-486c-bbc6-d4c436df0d63",
+    "apiType": "demand_a",
+    "transactionType": "creation",
+    "submittedAt": "2023-05-09T15:04:19.422Z",
+    "updatedAt": "2023-05-09T15:04:37.227Z"
+  }
+]
+```
+
+Enhancement: The local id changes to `pending` at this point. For example, with Member A:
+```
+{
+  "id": "18f832d2-162f-486c-bbc6-d4c436df0d63",
+  "owner": "Member A",
+  "state": "pending",
+  "parametersAttachmentId": "78b6a81c-5ef3-4549-82a9-e9459ca0cc93",
+  "createdAt": "2023-05-09T14:58:17.408Z",
+  "updatedAt": "2023-05-09T14:58:17.408Z"
+}
+```
+
+The indexers on Member B and Optimiser's dscp-matchmaker-api will process the new `demand_a` when the block containing it is finalised, and the new `demand_a` will be visible to them using `GET /v1/demandA`.
+
+#### Member B
+Inputting the `updatedSince` as: `2023-05-09T14:04:37.227Z`
+
+Response:
+```
+[
+  {
+    "id": "d51277a0-1e23-4d09-aceb-c10ecc147560",
+    "owner": "Member A",
+    "state": "created",
+    "parametersAttachmentId": "da003d33-3f0f-4c18-9ccb-5d122a7176ff",
+    "createdAt": "2023-05-09T15:04:37.279Z",
+    "updatedAt": "2023-05-09T15:04:37.279Z"
+  }
+]
+```
+
+#### Optimiser
+
+Inputting the `updatedSince` as: `2023-05-09T14:04:37.227Z`
+
+Response:
+```
+[
+  {
+    "id": "d51277a0-1e23-4d09-aceb-c10ecc147560",
+    "owner": "Member A",
+    "state": "created",
+    "parametersAttachmentId": "da003d33-3f0f-4c18-9ccb-5d122a7176ff",
+    "createdAt": "2023-05-09T15:04:37.279Z",
+    "updatedAt": "2023-05-09T15:04:37.279Z"
+  }
+]
+```
+
+Member B creates `demand_b` in a similar manner to creating `demand_a`, including the parameters file for `demand_b`.
+
+#### POST /v1/attachment
+
+Response:
+```
+{
+  "id": "18d3b77a-ded6-4fb5-ae16-edd8c1ba1a92",
+  "filename": "attachment_capacity.json",
+  "size": 1304,
+  "createdAt": "2023-05-09T15:32:43.664Z"
+}
+```
+
+#### POST request to /v1/demandB
+
+Response`:
+```
+{
+  "id": "566f6819-eb37-4102-9cfb-1c79c03cdb9b",
+  "owner": "Member B",
+  "state": "created",
+  "parametersAttachmentId": "18d3b77a-ded6-4fb5-ae16-edd8c1ba1a92",
+  "createdAt": "2023-05-09T15:43:45.427Z",
+  "updatedAt": "2023-05-09T15:43:45.427Z"
+}
+```
+
+When Member B is ready for `demand_b` to exist on the blockchain, they send a `POST request to /v1/demandB/{demandBId}/creation`.
+
+Response:
+```
+{
+  "id": "cce98ffb-9e88-4876-ab99-b4ae45f589a9",
+  "state": "submitted",
+  "localId": "566f6819-eb37-4102-9cfb-1c79c03cdb9b",
+  "apiType": "demand_b",
+  "transactionType": "creation",
+  "submittedAt": "2023-05-09T15:45:32.313Z",
+  "updatedAt": "2023-05-09T15:45:32.313Z"
+}
+```
+
+Optimiser creates a `match2` by sending a `POST request to /v1/match2/{match2Id}/proposal`, including the ids for `demand_a` and `demand_b`. From the optimiser node, send a `GET /v1/demandA` & `GET /v1/demandB` to retrieve the id’s:
```

Comment:
initial POST url is ` /v1/match2`

Reply:
```suggestion
}
```

Optimiser creates a `match2` by sending a `POST request to /v1/match2`, including the ids for `demand_a` and `demand_b`. From the optimiser node, send a `GET /v1/demandA` & `GET /v1/demandB` to retrieve the id’s:
```

Reply:
```suggestion
Optimiser creates a `match2` by sending a `POST request to /v1/match2`, including the ids for `demand_a` and `demand_b`. From the optimiser node, send a `GET /v1/demandA` & `GET /v1/demandB` to retrieve the id’s:
```